src/js/utils/DSLUpdateUtil.js

```diff
@@ -3,11 +3,74 @@ import DSLCombinerTypes from '../constants/DSLCombinerTypes';
 import DSLExpression from '../structs/DSLExpression';
 import DSLFilterTypes from '../constants/DSLFilterTypes';
 import DSLUtil from './DSLUtil';
-import {FilterNode} from '../structs/DSLASTNodes';
+
+const STRING_EXPR = /(['"])[^\1]+?(?=\1)\1/;
 
 const DSLUpdateUtil = {
 
   /**
+   * Default function used by the `applyAdd` function to detect is similar node
+   * already exists in the expression.
+   *
+   * @param {FilterNode} referenceNode - The node to compare against
+   * @param {FilterNode} compareNode - The node to compare
+   * @returns {Boolean} Returns `true` if there is no similar node in the AST
+   */
+  defaultNodeCompareFunction(referenceNode, compareNode) {
+    if (referenceNode.filterType !== compareNode.filterType) {
+      return false;
+    }
+
+    if (referenceNode.filterType !== DSLFilterTypes.ATTRIB) {
+      return true;
+    }
+
+    return referenceNode.filterParams.label === compareNode.filterParams.label;
+  },
+
+  /**
+   * Clean-up a DSL expression without damaging it's semantic meaning.
+   * This function does the following:
+   *
+   * - Expands single parenthesis (ex. "foo (bar) baz")
+   * - Trims consecutive whitespaces (ex "foo  bar ")
+   * - Trims consecutive or empty commas (ex. "foo,, bar, , baz")
+   * - Removes orphan comas (ex. ", foo (, bar) label:,baz")
+   *
+   * @param {String} src - The source expression string
+   * @returns {String} The cleaned-up expression string
+   */
+  cleanupExpressionString(src) {
+    let strings = [];
+
+    // Extract string expressions so we have a simpler expression to work with
+    src = src.replace(STRING_EXPR, (match) => {
+      strings.push(match);
+      return '\x01';
+    });
+
+    // Perform clean-ups
+    src = src.replace(/\(([^ ,]+)\)/g, '$1'); // Single parenthesis
+    src = src.replace(/\s+/g, ' '); // Consecutive spaces
+    src = src.replace(/,(\s*,)+/g, ','); // Consecutive commas
+    src = src.replace(/^\s*,/g, ''); // Orphan commas (left-side)
+    src = src.replace(/,\s*$/g, ''); // Orphan commas (right-side)
+    src = src.replace(/\(\s*,\s*/g, '('); // Orphan commas (left-paren)
+    src = src.replace(/\s*,\s*\)/g, ')'); // Orphan commas (right-paren)
+    src = src.replace(/:,+/g, ':'); // Orphan commas (multi-value)
+    src = src.replace(/\s+,\s+/g, ', '); // Commas with surrounding whitespace
+
+    // Put strings back
+    /* eslint-disable no-control-regex */
+    src = src.replace(/\x01/g, () => {
```

Comment:
Why do we need this? I mean why do we enable rules if we disable them somewhere in the code?